File: semana_6/Ejercicios/2-EmpleadosEmpresa/empleadosEmpresa.py


class Empleado():

    def __init__(self,nombre,edad,salario):
        self.nombre = nombre
        self.edad = edad
        self.salario = salario

class Gerente(Empleado):
    def __init__(self,nombre,edad,salario,liderazgo,grupos):
        super().__init__(nombre,edad,salario)
        self.liderazgo = liderazgo
        self.grupos = grupos

    def describir_rol(self):
        print(f"Gerente\nNombre:{self.nombre}\nSalario:{self.salario}\nLiderazgo:{self.liderazgo}\nGrupos:{self.grupos}")
        
    def MostrarManejoDeGrupos(self):
        print(f"grupos: {self.grupos}")
        
    def SetLiderazgo(self,nuevoLiderazgo):
        self.liderazgo = nuevoLiderazgo


class Ingeniero(Empleado):
    def __init__(self,nombre,edad,salario,tecnicatura):
        super().__init__(nombre,edad,salario)
        self.tecnicatura = tecnicatura
        self.habilidades = []
    
    def describir_rol(self):
        print(f"Ingeniero\nNombre:{self.nombre}\nSalario:{self.salario}\nTecnicatura:{self.tecnicatura}\nHabilidades:{self.habilidades}")
        
    
    def AgregarHabilidar(self,nuevaHabilidad):
        self.habilidades.append(nuevaHabilidad)
               


class Asistente(Empleado):
    def __init__(self,nombre,edad,salario):
        super().__init__(nombre,edad,salario)
        self.tareas = []
        self.citas = []
        
    def describir_rol(self):
        print(f"Asistente\nNombre:{self.nombre}\nSalario:{self.salario}\nTareas:{self.tareas}\nCitas:{self.citas}")
        
    
    def MostrarTareas(self):
        print(f"Tareas:\n{self.tareas}")
        
    def AgregarTarea(self,nuevaTarea):
        self.tareas.append(nuevaTarea)
    
    def EliminarTarea(self,tarea):
        self.tareas.remove(tarea)
        
    def AgregarCita(self,nuevaCita):
        self.citas.append(nuevaCita)
    
    


    

File: semana_6/Ejercicios/2-EmpleadosEmpresa/empleadosEmpresaCPP/empleadosEmpresa.cpp
#include <iostream>
#include <vector>

class Empleado {
    public:
    std::string nombre;
    int edad;
    int salario;

    Empleado(std::string _nombre, int _edad,int _salario) {
        nombre = _nombre;
        edad = _edad;
        salario = _salario;
    }

    void describir_rol() {
        std::cout << "nombre: " << nombre << std::endl;
        std::cout << "edad: " << edad << std::endl;
        std::cout << "salario: " << salario << std::endl; 
    }

};

class Gerente : public Empleado {

    public:
    std::string liderazgo;
    std::vector<std::string> grupos;


    Gerente(std::string _nombre, int _edad, int _salario,std::string _liderazgo) : Empleado(_nombre,_edad,_salario), grupos() {
        liderazgo = _liderazgo;

    }

    void describir_rol() {
        std::cout << "nombre: " << nombre << std::endl;
        std::cout << "edad: " << edad << std::endl;
        std::cout << "salario: " << salario << std::endl; 
        std::cout << "liderazgo: " << liderazgo << std::endl;
    }

};

class Ingeniero  : public Empleado {

    public:
    std::string tecnicatura;
    std::vector<std::string> habilidades;


    Ingeniero (std::string _nombre, int _edad, int _salario,std::string _tecnicatura)
    : Empleado(_nombre,_edad,_salario), habilidades() {
        _tecnicatura = tecnicatura;

    }

    void describir_rol() {
        std::cout << "nombre: " << nombre << std::endl;
        std::cout << "edad: " << edad << std::endl;
        std::cout << "salario: " << salario << std::endl; 
        std::cout << "liderazgo: " << tecnicatura << std::endl;
    }

};


class Asistente  : public Empleado {

    public:
    std::vector<std::string> tareas;
    std::vector<std::string> citas;


    Asistente (std::string _nombre, int _edad, int _salario) : Empleado(_nombre,_edad,_salario), tareas(), citas() {

    }

    void describir_rol() {
        std::cout << "nombre: " << nombre << std::endl << "edad: " << edad << std::endl << "salario: " << salario << std::endl;
        for (int i = 0; i < tareas.size(); i++)
        {
            std::cout << "Tarea " << i << ":"<< tareas[i];
        }
    }

    void Agregar_tareas(std::string nuevaTareas) {
        tareas.push_back(nuevaTareas);

    }

};


int main() {

    Gerente kevin("kevin",19,3000,"fuerte");
    kevin.describir_rol();

    Asistente Can("Can",20,3000);
    Can.Agregar_tareas("Administrar");
    Can.describir_rol();



}